R/AzureDataLake.R

```diff
@@ -0,0 +1,368 @@
+        FileStatuses.FileStatus.modificationTime = character(0),
+        FileStatuses.FileStatus.replication = character(0),
+        FileStatuses.FileStatus.permission = character(0),
+        FileStatuses.FileStatus.owner = character(0),
+        FileStatuses.FileStatus.group = character(0),
+        FileStatuses.FileStatus.aclBit = character(0)
+      )
+    )
+  }
+  resDf <- as.data.frame(resJsonObj)
+  resDf
+}
+
+
+#' Azure Data Lake MKDIRS for specified relativePath of an azure data lake account.
+#'
+#' @inheritParams createAzureContext
+#' @inheritParams azureAuthenticate
+#'
+#' @param azureActiveContext Provide an `azureActiveContext` object used for authentication.
+#' @param azureDataLakeAccount Provide the name of the Azure Data Lake account.
+#' @param relativePath Provide a relative path of the directory.
+#' @param permission Provide the permission to be set for the directory.
+#' @param verbose Print tracing information (default FALSE).
+#'
+#' @return Returns a boolean.
+#'
+#' @template
+#' @references
+#' @family Azure Data Lake Store functions
+#' @export
+azureDataLakeMkdirs <- function(azureActiveContext, azureDataLakeAccount, relativePath, permission = NULL, verbose = FALSE) {
+
+  if (!missing(azureActiveContext) && !is.null(azureActiveContext)) {
+    assert_that(is.azureActiveContext(azureActiveContext))
+    azureCheckToken(azureActiveContext)
+  }
+  assert_that(is_storage_account(azureDataLakeAccount))
+  if (!missing(permission) && !is.null(permission)) assert_that(is_permission(permission))
+  verbosity <- set_verbosity(verbose)
+
+  URL <- paste0(
+    "https://", azureDataLakeAccount, ".azuredatalakestore.net/webhdfs/v1/",
+    relativePath,
+    "?op=MKDIRS",
+    "&api-version=2016-11-01"
+  )
+  if (!missing(permission) && !is.null(permission)) URL <- paste0(URL, "&permission=", permission)
+
+  resHttp <- callAzureDataLakeApi(URL, verb = "PUT",
+                                  azureActiveContext = azureActiveContext,
+                                  verbose = verbose)
+  if (status_code(resHttp) == 404) {
+    warning("Azure data lake response: resource not found")
+    return(NULL)
+  }
+  stopWithAzureError(resHttp)
+
+  resJsonStr <- content(resHttp, "text", encoding = "UTF-8")
+  resJsonObj <- jsonlite::fromJSON(resJsonStr)
+  resDf <- as.data.frame(resJsonObj)
+  resDf$boolean
+}
+
+#' Azure Data Lake CREATE for specified relativePath of an azure data lake account.
+#'
+#' @inheritParams createAzureContext
+#' @inheritParams azureAuthenticate
+#'
+#' @param azureActiveContext Provide an `azureActiveContext` object used for authentication.
+#' @param azureDataLakeAccount Provide the name of the Azure Data Lake account.
+#' @param relativePath Provide a relative path of the directory.
+#' @param overwrite Overwrite existing files (default FALSE).
+#' @param permission Provide the permission to be set for the directory.
+#' @param contents Provide contents to write to `relativePath`
```

Comment:
Any constraints on the contents? Type? Size? Are they checked in the code? Tested?

Reply:
Fixed in https://github.com/omkarksa/AzureSMR/pull/1 will raise a pull request once this is merged into master